octoprint_TimeToFilament/__init__.py

```diff
@@ -66,6 +79,33 @@ def on_startup(self, host, port):
   @octoprint.plugin.BlueprintPlugin.route("/get_settings_defaults", methods=["GET"])
   def get_settings_defaults_as_string(self):
     return json.dumps(self.get_settings_defaults())
+    
+  def is_blueprint_csrf_protected(self):
+    return True
+    
+  @octoprint.plugin.BlueprintPlugin.route("/", methods=["GET"])
+  def get_cache(self):
+    if self._printer.get_state_id() not in (["PRINTING", "PAUSED"]):
+      return flask.jsonify(None)
+    addData = self.additional_state_data(False)
```

Comment:
In python we prefer `snake_case` over `camelCase`, which is how Java does it.  I left some of the JSON keys in camelCase because that's how OctoPrint did it but I would prefer snake case, to be consistent with python's names guidelines.

https://stackoverflow.com/a/8908798/4454